src/libraries/System.IO.Pipes.AccessControl/tests/NamedPipeTests/NamedPipeServerStreamAclTests.cs

```diff
@@ -0,0 +1,316 @@
+        [InlineData(PipeAccessRights.ReadExtendedAttributes)]
+        [InlineData(PipeAccessRights.ReadAttributes)]
+        [InlineData(PipeAccessRights.ReadPermissions)]
+        [InlineData(PipeAccessRights.Write)]
+        [InlineData(PipeAccessRights.WriteExtendedAttributes)]
+        [InlineData(PipeAccessRights.WriteAttributes)]
+        public void Create_InvalidAdditionalAccessRights(PipeAccessRights additionalAccessRights)
+        {
+            // GetBasicPipeSecurity returns an object created with PipeAccessRights.ReadWrite as default. This enum is formed by:
+            //     - PipeAccessRights.Read: This enum is formed by:
+            //         - PipeAccessRights.ReadData | PipeAccessRights.ReadExtendedAttributes | PipeAccessRights.ReadAttributes | PipeAccessRights.ReadPermissions
+            //     - PipeAccessRights.Write: This enum is formed by:
+            //         - PipeAccessRights.WriteData | PipeAccessRights.WriteExtendedAttributes | PipeAccessRights.WriteAttributes
+
+            // additionalAccessRights gets bitwise merged with the 'dwOpenMode' parameter we pass to CreateNamedPipeW.
+            // This parameter can acquire any of the values described here: https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-createnamedpipea
+            // It's particularly important to mention that two of the accepted values collide with the value of two PipeAccessRights enum values:
+            // - ReadData (0x1): Same value as PIPE_ACCESS_INBOUND
+            // - WriteData (0x2): Same value as PIPE_ACCESS_OUTBOUND
+
+            // Any other value will throw with the message 'The parameter is incorrect.'
+            Assert.Throws<IOException>(() =>
+            {
+                Create_AdditionalAccessRights(additionalAccessRights).Dispose();
+            });
+        }
+
+        [Theory]
+        [InlineData(PipeAccessRights.CreateNewInstance)]
+        [InlineData(PipeAccessRights.Delete)]
+        public void Create_WindowsNotAcceptedAdditionalAccessRights(PipeAccessRights additionalAccessRights)
+        {
+            // Exception message: "The parameter is incorrect."
+            // Neither CreateNewInstance (0x4) nor Delete (0x10000) collide with any of the dwOpenMode values that get into the bitwise combination:
+            // PipeOptions, PipeDirection, Interop.Kernel32.FileOperations.FILE_FLAG_FIRST_PIPE_INSTANCE
+            // But Windows does not accept them anyway
+            Assert.Throws<IOException>(() =>
+            {
+                Create_AdditionalAccessRights(additionalAccessRights).Dispose();
+            });
+        }
+
+        [Fact]
+        public void Create_NotEnoughPrivilegesAdditionalAccessRights()
+        {
+            // Exception message: "A required privilege is not held by the client"
+            Assert.Throws<IOException>(() =>
+            {
+                Create_AdditionalAccessRights(PipeAccessRights.AccessSystemSecurity).Dispose();
```

Comment:
I think the reason why this is thrown is because the PipeSecurity object gets created using WellKnownSidType.BuiltinUsersSid. I am not sure if I should be adding unit tests creating PipeSecurity objects using other WellKnownSidType enum values. This is the one Jeremy recommended me to use.